Task #3 - import required packages with their customary aliases

In [2]:
import pandas as pd

In [3]:
import numpy as np

In [4]:
import matplotlib.pyplot as plt

In [5]:
import seaborn as sns

Task #4 Using the pandas `read_csv()` method, read the GDP dataset into your notebook as a DataFrame called `gdp_df`. Take a look at the first few and last few rows to familiarize yourself with what is contained in this dataset.

In [6]:
gdp_df = pd.read_csv('..\data\gdp_percapita.csv')

In [7]:
print(gdp_df.head())

  Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2020  1970.560169              NaN
1     Afghanistan  2019  2065.036235              NaN
2     Afghanistan  2018  2033.804389              NaN
3     Afghanistan  2017  2058.400221              NaN
4     Afghanistan  2016  2057.067978              NaN


In [8]:
print(gdp_df.tail())

     Country or Area  Year        Value  Value Footnotes
7171        Zimbabwe  1994  2603.927369              NaN
7172        Zimbabwe  1993  2420.092791              NaN
7173        Zimbabwe  1992  2437.152908              NaN
7174        Zimbabwe  1991  2733.696626              NaN
7175        Zimbabwe  1990  2652.129209              NaN


In [9]:
print(gdp_df.describe())

              Year          Value  Value Footnotes
count  7176.000000    7176.000000              0.0
mean   2005.405936   16958.938497              NaN
std       8.853043   19243.365228              NaN
min    1990.000000     436.720371              NaN
25%    1998.000000    3591.252337              NaN
50%    2006.000000    9990.901201              NaN
75%    2013.000000   23478.750492              NaN
max    2020.000000  161971.464661              NaN


In [10]:
print(gdp_df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7176 entries, 0 to 7175
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7176 non-null   object 
 1   Year             7176 non-null   int64  
 2   Value            7176 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 224.4+ KB
None


Q5. How many rows and columns are in `gdp_df`? What are the data types of each column?

A5. there are 7176 rows since python starts everything with zero, 4 columns (not including the number assigned to the row), two floats, one column of integers, one column labeled object (text or string)

6. Drop the `Value Footnotes` column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [11]:
gdp_df=gdp_df.drop('Value Footnotes',axis=1)
gdp_df=gdp_df.rename(columns={'Country or Area':'Country', 'Value':'GDP_Per_Capita'})

In [12]:
print(gdp_df.head())

       Country  Year  GDP_Per_Capita
0  Afghanistan  2020     1970.560169
1  Afghanistan  2019     2065.036235
2  Afghanistan  2018     2033.804389
3  Afghanistan  2017     2058.400221
4  Afghanistan  2016     2057.067978


Q7. How many countries have data for all years? Which countries are missing many years of data? Look at the number of observations per year. What do you notice? 

In [13]:
country_count=gdp_df.groupby('Country')['GDP_Per_Capita'].count()

In [14]:
print(country_count)

Country
Afghanistan                    19
Africa Eastern and Southern    31
Africa Western and Central     31
Albania                        31
Algeria                        31
                               ..
Vietnam                        31
West Bank and Gaza             27
World                          31
Zambia                         31
Zimbabwe                       31
Name: GDP_Per_Capita, Length: 242, dtype: int64


In [15]:
country_count=gdp_df.groupby('Country')['GDP_Per_Capita'].count()

In [17]:
country_count = country_count.sort_values(ascending=False)

In [19]:
print(country_count)

Country
Least developed countries: UN classification          31
Niger                                                 31
Middle East & North Africa (excluding high income)    31
Middle income                                         31
Mongolia                                              31
                                                      ..
Kosovo                                                13
Turks and Caicos Islands                              10
Sint Maarten (Dutch part)                             10
Somalia                                                8
Djibouti                                               8
Name: GDP_Per_Capita, Length: 242, dtype: int64


Q7a. How Many Countries have data for all years? 
A7a. (205)

In [24]:
all_years=country_count[country_count==31]
count_allyears = len(all_years)

In [26]:
print(count_allyears)

205


In [20]:
missing_data=country_count[country_count<30]

In [21]:
print(missing_data)

Country
Slovak Republic              29
Cambodia                     28
West Bank and Gaza           27
Bosnia and Herzegovina       27
Maldives                     26
Moldova                      26
Kuwait                       26
Iceland                      26
Latvia                       26
Croatia                      26
Israel                       26
Slovenia                     26
Lithuania                    26
Estonia                      26
Serbia                       26
Canada                       24
San Marino                   24
Small states                 24
Montenegro                   24
Libya                        22
Palau                        21
Liberia                      21
Other small states           21
Qatar                        21
Curaçao                     21
Timor-Leste                  21
São Tomé and Principe      20
Afghanistan                  19
Nauru                        17
Cayman Islands               15
Kosovo                       13
